File: app/src/main/java/com/joyhonest/sports_camera/boot_Activity.java
package com.joyhonest.sports_camera;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

public class boot_Activity extends AppCompatActivity {
    @Override
    public void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        setContentView(R.layout.activity_boot_);
        getWindow().getDecorView().setBackgroundResource(R.mipmap.boot_interface);
        MyApp.F_makeFullScreen(this);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (!AppPrefs.getInstance(getApplicationContext()).getPrivacyPolicyAgreement()) {
                    requestPrivacyPolicyAgreement();
                    return;
                }
                startActivity(new Intent(boot_Activity.this, SplashActivity.class));
                finish();
                overridePendingTransition(0, 0);
            }
        }, 1500L);
    }

    @Override
    public void onResume() {
        super.onResume();
        MyApp.F_makeFullScreen(this);
    }

    public void requestPrivacyPolicyAgreement() {
        PrivacyPolicyAgreementDialog newInstance = PrivacyPolicyAgreementDialog.newInstance(this, getResources().getString(R.string.privacy_introduce_title), getResources().getString(R.string.privacy_introduce_content), getResources().getString(R.string.privacy_agree), getResources().getString(R.string.privacy_disagree), new PrivacyPolicyAgreementDialog.Listener() {
            @Override
            public void onYes() {
                AppPrefs.getInstance(getApplicationContext()).setPrivacyPolicyAgreement(true);
                startActivity(new Intent(boot_Activity.this, SplashActivity.class));
                finish();
                overridePendingTransition(0, 0);
            }

            @Override
            public void onNo() {
                finish();
            }
        });
        newInstance.setCancelable(false);
        newInstance.show(getSupportFragmentManager(), "Privacy Policy Agreement");
    }
}

File: app/src/main/java/com/joyhonest/sports_camera/boot_Activity.kt
package com.joyhonest.sports_camera

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity

class boot_Activity : AppCompatActivity() {
    public override fun onCreate(bundle: Bundle?) {
        super.onCreate(bundle)
        setContentView(R.layout.activity_boot_)
        window.decorView.setBackgroundResource(R.mipmap.boot_interface)
        MyApp.F_makeFullScreen(this)
        Handler().postDelayed(Runnable {
            if (!AppPrefs.getInstance(applicationContext).privacyPolicyAgreement) {
                requestPrivacyPolicyAgreement()
                return@Runnable
            }
            startActivity(Intent(this@boot_Activity, SplashActivity::class.java))
            finish()
            overridePendingTransition(0, 0)
        }, 1500L)
    }

    public override fun onResume() {
        super.onResume()
        MyApp.F_makeFullScreen(this)
    }

    fun requestPrivacyPolicyAgreement() {
        val newInstance: PrivacyPolicyAgreementDialog =
                PrivacyPolicyAgreementDialog.newInstance(
                        this,
                        resources.getString(R.string.privacy_introduce_title),
                        resources.getString(R.string.privacy_introduce_content),
                        resources.getString(R.string.privacy_agree),
                        resources.getString(R.string.privacy_disagree),
                        object : PrivacyPolicyAgreementDialog.Listener {
                            override fun onYes() {
                                AppPrefs.getInstance(applicationContext).privacyPolicyAgreement = true
                                startActivity(Intent(this@boot_Activity, SplashActivity::class.java))
                                finish()
                                overridePendingTransition(0, 0)
                            }

                            override fun onNo() {
                                finish()
                            }
                        })
        newInstance.isCancelable = false
        newInstance.show(supportFragmentManager, "Privacy Policy Agreement")
    }
}